src/lib/db/suggest-change-store.ts

```diff
@@ -0,0 +1,293 @@
+            )
+            .leftJoin(`users as eventUser`, 'eventUser.id', 'event.createdBy')
+            .select(
+                'changeSet.secret',
+                'changeSet.environment',
+                'changeSet.project',
+                'changeSet.createdAt',
+                'changeSetUser.username as changeSetUsername',
+                'changeSetUser.imageUrl as changeSetAvatar',
+                'change.id as changeId.',
+                'change.action as changeAction',
+                'change.payload as changePayload',
+                'change.created_at as changeCreatedAt',
+                'changUser.username as changeCreatedByUsername',
+                'changUser.imageUrl as changeCreatedByAvatar',
+                'event.id as eventId',
+                'event.event as eventType',
+                'event.data as eventData',
+                'event.created_at as eventCreatedAt',
+                'eventUser.username as eventCreatedByUsername',
+                'eventUser.imageUrl as eventCreatedByAvatar',
+            );
+    };
+
+    getAll = async (): Promise<ISuggestChangeset[]> => {
+        const rows = await this.buildSuggestChangeSetChangesEventsQuery();
+        return this.mapRows(rows);
+    };
+
+    getForProject = async (project: string): Promise<ISuggestChangeset[]> => {
+        const rows = await this.buildSuggestChangeSetChangesEventsQuery().where(
+            {
+                project,
+            },
+        );
+
+        return this.mapRows(rows);
+    };
+
+    getForEnvironment = async (
+        environment: string,
+    ): Promise<ISuggestChangeset[]> => {
+        const rows = await this.buildSuggestChangeSetChangesEventsQuery().where(
+            {
+                environment,
+            },
+        );
+
+        return this.mapRows(rows);
+    };
+
+    get = async (id: number): Promise<ISuggestChangeset> => {
+        const rows = await this.buildSuggestChangeSetChangesEventsQuery().where(
+            {
+                id,
+            },
+        );
+
+        return this.mapRows(rows)[0];
+    };
+
+    create = async (
+        suggestChangeSet: PartialSome<ISuggestChangeset, 'id'>,
+        user: Partial<Pick<User, 'username' | 'email'>>,
+    ): Promise<ISuggestChangeset> => {
+        const [{ id }] = await this.db(T.SUGGEST_CHANGE_SET)
+            .insert<ISuggestChangesetInsert>({
+                id: suggestChangeSet.id,
+                environment: suggestChangeSet.environment,
+                state: suggestChangeSet.state,
+                project: suggestChangeSet.project,
+                changes: JSON.stringify(suggestChangeSet.changes),
+                created_at: suggestChangeSet.createdAt,
+                created_by: user.username || user.email,
+            })
+            .returning('id');
+
+        suggestChangeSet.changes.forEach((change) => {
+            this.addChangeToSet(change, id, user);
+        });
+
+        return this.get(id);
```

Comment:
I think we should return the entire changeSet object from the create, this will allow the frontend to save a call and use the returning value instead.